### PR TITLE
nxodejs: fix cross

### PR DESCRIPTION
**Pull request by Cogitri**

### CODE REVIEW

**Comment by Cogitri** [comment]
aarch64 binaries tested on my RPI3

**Comment by Hoshpak** [comment]
there's a typo in the commit message. Besides, it still fails on armv6l and armv7l.

**Comment by Cogitri** [comment]
> Besides, it still fails on armv6l and armv7l.

Hum, seems like V8 assumes that we build for 32-bit if we compile to a 32-bit platform

FWIW, we have to do this due to https://github.com/nodejs/node/pull/21079, their code generation tool links against V8 so we have to build it for the host too (*sigh*)

**Comment by Cogitri** [comment]
https://github.com/nodejs/node/blob/master/deps/v8/src/base/build_config.h#L149

Seems like we actually have to build nodejs >= 10.9.0 from a i686 host...

**Comment by Hoshpak** [comment]
Since all builders are x86_64 and that is not likely to change (especially not to a legacy architecture like i686), that means we just cannot build nodejs for 32 bit arm architectures any more.

**Comment by Cogitri** [comment]
Yup, currently marking it as broken

**Comment by Cogitri** [comment]
Alright, this should work on aarch64* now, but I'll do some more testing - CI and builders seem busy anyway :P

**Comment by Hoshpak** [comment]
this will also require removal of nodejs from the armv[67]l repositories as the builders are currently staging because of nodejs.

**Comment by Cogitri** [comment]
I don't think node >=10 has that option anymore and looking  at https://github.com/void-linux/void-packages/pull/6573/files#diff-8421a2e1c1611cb3b4ee616adbdd6a15L38 it seems like we did pass that to node.

File: srcpkgs/nodejs/template [hunk 1]
```diff
@@ -32,10 +35,11 @@ provides="nodejs-runtime-0_1"
 do_configure() {
 	local _args
 
-	export LD="$CXX"
+	export LD="$CC"
+	export LDXX="$CXX"
```
Comment (maxice8) [inline srcpkgs/nodejs/template]: LDXX ?

Comment (Cogitri) [reply srcpkgs/nodejs/template]: Yup, node uses that too, e.g. here: https://github.com/nodejs/node/blob/8b4af64f50c5e41ce0155716f294c24ccdecad03/tools/gyp/pylib/gyp/generator/ninja.py

I guess we don't need it for `make` though right now.

**Comment by maxice8** [comment]
Please notify when proposed changes are made

**Pull request merged**